### PR TITLE
[full ci] Set ChangeVersion

### PR DESCRIPTION
**Pull request by caglar10ur**
This property is useful because it guards against updates that have happened
between when the VM’s config is read and when it is applied.

Will return "Cannot complete operation due to concurrent modification by another
operation.." on failure.

Right now we are calling Refresh() in every stage, following PRs will
remove that and introduce retry logic.

### CODE REVIEW

**Comment by dougm** [comment]
lgtm

File: lib/portlayer/exec/container.go [hunk 1]
```diff
@@ -258,6 +272,11 @@ func (c *Container) Commit(ctx context.Context, sess *session.Session, h *Handle
 
 		c.State = *h.State
 		commitEvent = events.ContainerStarted
+
+		// refresh the struct with what propery collector provides
+		if err := c.Refresh(); err != nil {
```
Comment (emlin) [inline lib/portlayer/exec/container.go]: The refresh happens after reconfigure. Does that mean if there is inconsistency between portlayer and VC, the operation will always fail?
Currently, the VM power on will change the changeVersion, so out of band operation might cause failure with it.

Comment (caglar10ur) [inline lib/portlayer/exec/container.go]: True out of band operations will cause failure. That's why we will introduce retry logic to the personality.

**Comment by emlin** [comment]
lgtm

**Comment by cgtexmex** [comment]
What's the benefit of having the new container attributes of Config & Runtime and the new Refresh method vs. the modifying the existing update method and utilizing the existing vm type?

**Comment by caglar10ur** [comment]
@cgtexmex to be honest I didn't want to touch Update as there is no comment what it supposed to be doing and why. Looks like it got introduces via 4e5b80197a0dda731c75274563a164d17fac7af8
 and since than that's the only caller

**Comment by cgtexmex** [comment]
lgtm

**Pull request merged**